src/Microsoft.AspNetCore.Mvc.Core/Internal/VirtualFileResultExecutor.cs

```diff
@@ -59,14 +69,31 @@ public Task ExecuteAsync(ActionContext context, VirtualFileResult result)
                         count: null,
                         cancellation: default(CancellationToken));
                 }
-                else
+                else if (range == null || !fileStream.CanSeek)
```

Comment:
Even if you can't seek, you can still scan forward in the stream to get to a subrange.